packages/cw721/src/msg.rs

```diff
@@ -21,15 +60,931 @@ pub enum Cw721ExecuteMsg {
         expires: Option<Expiration>,
     },
     /// Remove previously granted Approval
-    Revoke { spender: String, token_id: String },
+    Revoke {
+        spender: String,
+        token_id: String,
+    },
     /// Allows operator to transfer / send any token from the owner's account.
     /// If expiration is set, then this allowance has a time/height limit
     ApproveAll {
         operator: String,
         expires: Option<Expiration>,
     },
     /// Remove previously granted ApproveAll permission
-    RevokeAll { operator: String },
+    RevokeAll {
+        operator: String,
+    },
+
+    /// Mint a new NFT, can only be called by the contract minter
+    Mint {
+        /// Unique ID of the NFT
+        token_id: String,
+        /// The owner of the newly minter NFT
+        owner: String,
+        /// Universal resource identifier for this NFT
+        /// Should point to a JSON file that conforms to the ERC721
+        /// Metadata JSON Schema
+        token_uri: Option<String>,
+        /// Any custom extension used by this contract
+        extension: TNftExtensionMsg,
+    },
+
     /// Burn an NFT the sender has access to
-    Burn { token_id: String },
+    Burn {
+        token_id: String,
+    },
+
+    /// Custom msg execution. This is a no-op in default implementation.
+    Extension {
+        msg: TExtensionMsg,
+    },
```

Comment:
Change generic `TNftExtensionMsg` -> `TExtensionMsg`

Default implementation in trait is a no-op for `TExtensionMsg`. Custom contracts may override and provide custom extension msg.

`TNftExtensionMsg` is used `Mint` and `UpdateNftInfo` msgs:

```rs
...
    /// Mint a new NFT, can only be called by the contract minter
    Mint {
        /// Unique ID of the NFT
        token_id: String,
        /// The owner of the newly minter NFT
        owner: String,
        /// Universal resource identifier for this NFT
        /// Should point to a JSON file that conforms to the ERC721
        /// Metadata JSON Schema
        token_uri: Option<String>,
        /// Any custom extension used by this contract
        extension: TNftExtensionMsg,
    },
...
    /// The creator is the only one eligible to update NFT's token uri and onchain metadata (`NftInfo.extension`).
    /// NOTE: approvals and owner are not affected by this call, since they belong to the NFT owner.
    UpdateNftInfo {
        token_id: String,
        token_uri: Option<String>,
        extension: TNftExtensionMsg,
    },
...
```

Reply:
Nit on naming, `TExtensionMsg` sounds really generic. Isn't this just used for metadata? Maybe `TNftMetadataExtension`? Perhaps someone else has a better idea.